lib/plugins/query.js

```diff
@@ -22,14 +39,33 @@ function queryParser(options) {
     }
     assert.object(options, 'options');
 
+    /*
+     * Releases of restify 4.x up to 4.1.1 used qs@3 which effectively defaulted
+     * to `plainObjects=true` and `allowDots=true`. To maintain backward
+     * compatibility for the restify 4.x stream while using the latest qs
+     * version, we need to maintain those defaults. Note that restify-plugins
+     * changes back to the pre-restify-4.x behaviour. See test/query.test.js
+     * for more details.
+     */
+    var qsOptions = {
+        plainObjects: true,
+        allowDots: true
+    };
+    Object.keys(EXPOSED_QS_OPTIONS).forEach(function (k) {
```

Comment:
Not sure about the performance impact of Object.keys, but presumably you could use https://www.npmjs.com/package/lodash.foreach to optimize this.